src/objects.js

```diff
@@ -1,42 +1,70 @@
 const createPerson = (name, age) => {
-  // your code here
+  const person = {name:name, age:age};
+  return person;
 };
 
 const getName = object => {
-  // your code here
+  return object.name;
 };
 
 const getProperty = (property, object) => {
-  // your code here
+  return object[property];
 };
 
 const hasProperty = (property, object) => {
-  // your code here
+  const keys = Object.keys(object);
+  for (let i = 0; i < keys.length; i++) {
+    if (keys[i] === property.toString()) {
+      return true;
+    }
+  }
+  return false;
 };
 
 const isOver65 = person => {
-  // your code here
+  return person.age > 65;
 };
 
 const getAges = people => {
-  // your code here
+  const ages = [];
+  for (let i = 0; i < people.length; i++) {
+    ages.push(people[i].age);
+  }
+  return ages;
 };
 
 const findByName = (name, people) => {
-  // your code here
+  let match;
+  for (let i = 0; i < people.length; i++) {
+    if (people[i].name === name) {
+      match = people[i];
+      break;
+    }
+  }
+  return match;
 };
 
 const findHondas = cars => {
-  // your code here
+  return cars.filter(car => car.manufacturer === 'Honda');
 };
 
 const averageAge = people => {
-  // your code here
+  if (people.length === 0) {
+    return 0;
+  }
+  const avgAge = people.reduce((acc, person) => acc + person.age, 0);
+  return avgAge / people.length;
 };
 
 const createTalkingPerson = (name, age) => {
-  // your code here
-};
+  return {
+    'name': name,
+    'age': age,
+    'introduce': otherPerson => {
+      return `Hi ${otherPerson}, my name is ${name} and I am ${age}!`;
+    }
+    }
```

Comment:
don't put object property names as strings if you aren't including special characters. Also, as your `name` and `age` properties are also the names of the variables you can just pass them straight to the object. i.e. write as
```
return {
  name,
  age,
  introduce: ...
}
```